src/derivers/shared.ts

```diff
@@ -462,10 +500,15 @@ type GenerateEntropyArgs = {
  * @param args - The arguments for generating entropy.
  * @param args.chainCode - The parent chain code bytes.
  * @param args.extension - The extension bytes.
+ * @param cryptographicFunctions - The cryptographic functions to use. If
+ * provided, these will be used instead of the built-in implementations.
  * @returns The generated entropy bytes.
  */
-export function generateEntropy({ chainCode, extension }: GenerateEntropyArgs) {
-  return hmac(sha512, chainCode, extension);
+export async function generateEntropy(
+  { chainCode, extension }: GenerateEntropyArgs,
```

Comment:
For non exported functions like this one, would it be cleaner to add `cryptographicFunctions` to the main args?

Reply:
Might even be an option for exported functions if not breaking

Reply:
I prefer having it separately, as the main args are actually related to generating entropy, and the functions are just to pass them around to whichever function needs them.